packages/platform-core/src/lock-file-reader/npm_lock_file_reader.ts

```diff
@@ -0,0 +1,59 @@
+import fsp from 'fs/promises';
+import path from 'path';
+import z from 'zod';
+import {
+  Dependencies,
+  LockFileContents,
+  LockFileReader,
+} from './lock_file_reader_factory';
+
+/**
+ * NpmLockFileReader is an abstraction around the logic used to read and parse lock file contents
+ */
+export class NpmLockFileReader implements LockFileReader {
+  getLockFileContentsFromCwd = async (): Promise<LockFileContents> => {
+    const dependencies: Dependencies = [];
+    const packageLockJsonPath = path.resolve(
+      process.cwd(),
+      'package-lock.json'
+    );
+
+    let jsonLockParsedValue: Record<string, unknown>;
+    try {
+      const jsonLockContents = await fsp.readFile(packageLockJsonPath, 'utf-8');
+      jsonLockParsedValue = JSON.parse(jsonLockContents);
+    } catch (error) {
+      // We failed to get lock file contents either because file doesn't exist or it is not parse-able
```

Comment:
if we move this to cli-core then we should `printer."log debug"` that we filed to get lock file content.

similarly in other versions of this impl.